spot-client/src/spot-tv/native-functions/native-controller/native-controller.js

```diff
@@ -0,0 +1,45 @@
+import { logger } from 'common/logger';
+
+/**
+ * Implements a controller that utilizes the event emitter of the native (Electron) process if available.
+ *
+ * Using it will enable SpotTVs running wrapped in Electron access OS-native functions.
+ */
+class NativeControntroller {
+    /**
+     * Some.
+     */
+    constructor() {
+        // eslint-disable-next-line
+        this.ipcRenderer = this.isAvailable() && __non_webpack_require__('electron').ipcRenderer;
+    }
+
+    /**
+     * Returns true if the native functions are available (electron env.), false otherwise (web env.).
+     *
+     * @returns {boolean}
+     */
+    isAvailable() {
+        return window && window.process && window.process.emit;
+    }
+
+    /**
+     * Sends a message to the native process.
+     *
+     * @param {string} command - The command identifyer.
+     * @param {Object} args - Args of the command, if any.
+     * @returns {void}
+     */
+    sendMessage(command, args) {
+        if (this.isAvailable()) {
+            this.ipcRenderer.send('native-command', {
+                command,
+                args
+            });
+        } else {
+            logger.log(`Native functions are not available to do '${command}'`);
+        }
+    }
+}
+
+export default new NativeControntroller();
```

Comment:
What do you think about electron adding a global native controller variable. That way we can do feature detection per feature based on what's on that global. I don't know if that's even still possible to do but it's what was done on a previous project I worked on.

Reply:
Ticket filed